### ISSUE TITLE
question_mark should trigger on Some-pattern-matching let else statement

### ISSUE THREAD

**Issue by jackssrt** [rust-lang/rust-clippy#17545]
### Summary

I think that `clippy::question_mark` should trigger for `let Some(_) = _ else { return Err(_); }`

### Lint Name

manual_ok_or , question_mark

### Reproducer

I tried this code:

```rust
let an_option: Option<i32> = None;
let Some(_something) = an_option else {
    return Err(1);
};
```

I expected to see this happen:

A warning being produced telling me to use `Option::ok_or` or `Option::ok_or_else`, replacing the snippet with this:

```rust
let an_option: Option<i32> = None;
let _something = an_option.ok_or(1)?;
```

Instead, this happened:

Nothing, the code made it into production :c

### Version

```text
rustc 1.99.0-nightly (771916f90 2026-08-08)
binary: rustc
commit-hash: 771916f9028e7fe56d2685f2c4f698de5d7d6a45
commit-date: 2026-08-08
host: x86_64-unknown-linux-gnu
release: 1.99.0-nightly
LLVM version: 23.1.0
```

**Comment by jackssrt** [comment]
A niche case where this would break code is if the `else` branch uses `await`, make sure to check for that. Or, maybe just make sure the else branch is simple, ie.
`Err(_)?;`
or
`return Err(_);`
but that would of course leave false negatives...

**Comment by Jarcho** [comment]
I don't think `manual_ok_or` is the best spot for this. It's basically a combination of `manual_ok_or` and `question_mark` and the latter makes more sense to fit this under.

**Comment by jackssrt** [comment]
Updated the issue, thank you for the input
